java/client/src/org/openqa/selenium/firefox/internal/Executable.java

```diff
@@ -123,22 +123,24 @@ private static File locateFirefoxBinaryFromSystemProperty() {
     if (binaryName == null)
       return null;
 
-    File binary = new File(binaryName);
-    if (binary.exists())
```

Comment:
this check should remain, but could be expanded to ensure 'binary' is also a File, not a directory